src/libraries/System.Drawing.Common/src/System/Drawing/GdiplusNative.Unix.cs

```diff
@@ -23,23 +19,21 @@ internal unsafe partial class Gdip
             // When set to false, where Carbon Drawing is used instead.
             // macOS users can force X11 by setting the SYSTEM_DRAWING_COMMON_FORCE_X11 flag.
             public static bool UseX11Drawable { get; } =
-                !RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ||
+                !OperatingSystem.IsMacOS() ||
                 Environment.GetEnvironmentVariable("SYSTEM_DRAWING_COMMON_FORCE_X11") != null;
 
             internal static IntPtr LoadNativeLibrary()
             {
-                var assembly = System.Reflection.Assembly.GetExecutingAssembly();
-
                 IntPtr lib = IntPtr.Zero;
-                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                if (OperatingSystem.IsMacOS())
                 {
-                    if (!NativeLibrary.TryLoad("libgdiplus.dylib", assembly, default, out lib))
+                    if (!NativeLibrary.TryLoad("libgdiplus.dylib", out lib))
```

Comment:
This was intentionally using the higher level `NativeLibrary` API so that the runtime tries to find the `ibgdiplus.dylib` library in .NET specific locations in addition to the default OS specific locations. The low-level API uses OS specific locations only.

Reply:
I see. I thought the high-level API would trigger the `NativeLibraryResolver` but we were already in it; I checked the sources and realized it's not the case. I will revert it.